contracts/SignatureValidatorManager.sol

```diff
@@ -0,0 +1,222 @@
+     * @param signatureValidatorHooksAddress Address of the signature validator hooks contract
+     */
+    function setSignatureValidatorHooks(address signatureValidatorHooksAddress) external override {
+        if (signatureValidatorHooksAddress != address(0)) {
+            checkPermittedModule(signatureValidatorHooksAddress, MODULE_TYPE_SIGNATURE_VALIDATOR_HOOKS);
+
+            if (
+                !ISafeProtocolSignatureValidatorHooks(signatureValidatorHooksAddress).supportsInterface(
+                    type(ISafeProtocolSignatureValidatorHooks).interfaceId
+                )
+            ) revert ContractDoesNotImplementValidInterfaceId(signatureValidatorHooksAddress);
+        }
+        signatureValidatorHooks[msg.sender] = signatureValidatorHooksAddress;
+
+        emit SignatureValidatorHooksChanged(msg.sender, signatureValidatorHooksAddress);
+    }
+
+    /**
+     * @notice A non-view function that the Manager will call when an account has enabled this contract as a function handler in the Manager
+     * @param account Address of the account whose signature validator is to be used
+     * @param sender Address requesting signature validation
+     * @param data Calldata containing the 4 bytes function selector, 32 bytes message hash and payload.
+     *             Layout of data:
+     *             0x00 to 0x04 - 4 bytes function selector when this contract is set as a function handler in the SafeProtocolManager i.e. 0x1626ba7e
+     *             0x04 to 0x24 - 32 bytes hash of the message used for signing
+     *             0x24 to end - bytes containing signatures or signatureData either one of the below:
+     *             If first 4 bytes of signatureData are 0xb5c726cb i.e. bytes4(keccak256("Account712Signature(bytes32,bytes32,bytes)")); then it will be interpreted as follows:
+     *                  payload = abi.encodeWithSelector(0xb5c726cb, abi.encode(domainSeparator, structHash, signatures)
+     *                  Layout of `data` parameter in this case:
+     *                  0x00 to 0x04 - 4 bytes function selector when this contract is set as a function handler in the SafeProtocolManager i.e. 0x1626ba7e
+     *                  0x04 to 0x24 - 32 bytes hash of the signed message
+     *                  0x24 to 0x44 - 32 bytes offset to the start of `bytes` parameter
+     *                  0x44 to 0x64 - 32 bytes length of `bytes` parameter
+     *                  0x64 to 0x68 - 4 bytes of Signature selector
+     *                  0x68 to 0x88 - 32 bytes domain separator
+     *                  0x88 to 0xa8 - 32 bytes struct hash
+     *                  0xa8 to end - contains offset, length of bytes, and actual bytes containing signatures
+     *             Else:
+     *                 bytes containing signature data
+     *                 default validation flow will be used which will depend on the account implementation
+     *
+     */
```

Comment:
I didn't find this in the specs. Where is this encoding defined?

Just FYI created an issue for it in the specs https://github.com/safe-global/safe-core-protocol-specs/issues/56

Reply:
its here in the sample code: https://github.com/safe-global/safe-core-protocol-specs/blob/2bffd759dd12be5583594f302d97c35e0ab9fcf5/modules/README.md?plain=1#L233

Reply:
But it should be in the manager specs in this case, shouldn't it? If I want to implement my own manager, how do I learn this?

Reply:
Specifications are not user/dev documentation, it should include every detail of how the implementation should work

Reply:
> But it should be in the manager specs in this case, shouldn't it? If I want to implement my own manager, how do I learn this?

I propose that we put general info in function handler specs https://github.com/safe-global/safe-core-protocol-specs/tree/main/modules#function-handler.

The `handle` function says nothing on what must be the content of `data` param.
i.e. first 4 bytes must be the function selector followed by the parameter values

If we want to elaborate more specific case here of how `data` would look like in case of signature validator, I am fine with adding in validator specs.

Reply:
> If we want to elaborate more specific case here of how data would look like in case of signature validator, I am fine with adding in validator specs.

I'd expect the specs to be a little more elaborate on the general question: what kind of contract is a signature validation manager?

The answer would be something along the lines of:
A contract that does this and that allows setting a signature validation for an account, defines a `handle` function that can handle this type of arguments (encoding, etc), tracks the registry, etc, etc

Right now, the SignatureValidatorManager definition is mixed inside the definition of SignatureValidators (which is already fairly complex, and IMO, the Validator manager is a higher-level entity and should be defined separately)

Reply:
So I was mixing up things and seeing the manager as a spec-related thing, but in reality, it's an implementation detail: https://github.com/safe-global/safe-core-protocol-specs/issues/56

My bad